coverage_comment/template.py

```diff
@@ -133,3 +140,21 @@ def pct(val: decimal.Decimal | float) -> str:
         return f"{val.quantize(decimal.Decimal('0.01'), rounding=decimal.ROUND_DOWN).normalize():f}%"
     else:
         return f"{val:.0%}"
+
+
+def get_file_url_function(repo_name: str, pr_number: int) -> Callable:
+    def _(
+        filename: pathlib.Path,
+        line_start: int | None = None,
+        line_end: int | None = None,
+    ) -> str:
+        # To link to a file in a PR, GitHub uses the link to the file overview combined with a SHA256 hash of the file path
+        s = f"https://github.com/{repo_name}/pull/{pr_number}/files#diff-{hashlib.sha256(str(filename).encode('utf-8')).hexdigest()}"
+
+        if line_start is not None:
+            s += f"R{line_start}"
+        if line_end is not None and line_end > line_start:
```

Comment:
I'm not a fan that this function would crash if line_end was not None and line_start was. What about accepting `line_range` as `None | int | tuple[int, int]` ? (or even just tuple, it's unlikely that we'll ever call it with a single line number anyway)